llpc/context/llpcCompiler.cpp

```diff
@@ -778,7 +778,7 @@ Result Compiler::BuildShaderModule(const ShaderModuleBuildInfo *shaderInfo, Shad
 // @param shaderInfo : Shader info of this pipeline
 // @param [out] pipelineElf : Output Elf package
 Result Compiler::buildPipelineWithRelocatableElf(Context *context, ArrayRef<const PipelineShaderInfo *> shaderInfo,
-                                                 ElfPackage *pipelineElf) {
+                                                 ElfPackage *pipelineElf, StageCacheInfo *stageCacheAccesses) {
```

Comment:
You probably want to use an "ArrayRef" for this paramenter, and everywhere else it is passed internally.

Reply:
`ArrayRef` is only for referencing array-like continuous containers, e.g., `std::vector`, `llvm::SmallVector`, `std::array`, or plain C arrays of static size. Unlike the last two, `ArrayRef` doesn't have a size parameter in its type.

I'd recommend using `std::array` in the place where the source array is defined, and then passing it by `ArrayRef` or `std::array` reference.

Reply:
And `MutableArrayRef` is a mutable version of `ArrayRef` that allows to modify the underlying data.